Failure to build 1.4.0 on existing projects that were using 1.3.2

It seems that `1.4.0` is making assumptions about items in the current space, which is a breaking change, and fails to build on projects who were dependent on any `1` version, such as previously `1.3.2`.

I discovered this error because projects using new projects created using the `soroban-sdk` crate started pulling in `arbitrary` and `derive_arbitrary` `1.4.0` and see build errors.

```
$ git clone https://github.com/stellar/rs-soroban-sdk
$ git checkout fc32690edd223fd01054c625d9c6dca01bd99e67 # main at today's date
$ cd rs-soroban-sdk
$ cargo update arbitrary --precise 1.4.0
$ cargo test
...
error[E0107]: type alias takes 1 generic argument but 2 generic arguments were supplied
   --> /Users/leighmcculloch/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-xdr-22.0.0-rc.1.1/src/curr/generated.rs:935:42
    |
935 | #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
    |                                          ^^^^^^^^^ expected 1 generic argument
    |
note: type alias defined here, with 1 generic parameter: `T`
   --> /Users/leighmcculloch/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-xdr-22.0.0-rc.1.1/src/curr/generated.rs:226:6
    |
226 | type Result<T> = core::result::Result<T, Error>;
    |      ^^^^^^ -
    = note: this error originates in the derive macro `Arbitrary` (in Nightly builds, run with -Z macro-backtrace for more info)
...
error[E0053]: method `try_size_hint` has an incompatible type for trait
   --> /Users/leighmcculloch/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stellar-xdr-22.0.0-rc.1.1/src/curr/generated.rs:935:42
    |
935 | #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
    |                                          ^^^^^^^^^ expected `MaxRecursionReached`, found `generated::Error`
    |
    = note: expected signature `fn(_) -> std::result::Result<_, MaxRecursionReached>`
               found signature `fn(_) -> std::result::Result<_, generated::Error>`
    = note: this error originates in the derive macro `Arbitrary` (in Nightly builds, run with -Z macro-backtrace for more info)
...
...
```

Yep this broke our CI for us too

```
error[E0658]: use of unstable library feature 'error_in_core'
  --> /usr/local/cargo/registry/src/index.crates.io-6f17d22bba15001f/arbitrary-1.4.0/src/lib.rs:52:6
   |
52 | impl core::error::Error for MaxRecursionReached {}
   |      ^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #103765 <https://github.com/rust-lang/rust/issues/103765> for more information

For more information about this error, try `rustc --explain E06[58](https://github.com/georust/geo/actions/runs/11686445121/job/32542225331?pr=1258#step:7:59)`.
error: could not compile `arbitrary` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

--- a/derive/src/lib.rs
+++ b/derive/src/lib.rs
@@ -397,7 +397,7 @@ fn gen_size_hint_method(input: &DeriveInput) -> Result<TokenStream> {
                 }
 
                 #[inline]
-                fn try_size_hint(depth: usize) -> Result<(usize, ::core::option::Option<usize>), arbitrary::MaxRecursionReached> {
+                fn try_size_hint(depth: usize) -> ::core::result::Result<(usize, ::core::option::Option<usize>), arbitrary::MaxRecursionReached> {
                     arbitrary::size_hint::try_recursion_guard(depth, |depth| #hint)
                 }
             }
@@ -422,7 +422,7 @@ fn gen_size_hint_method(input: &DeriveInput) -> Result<TokenStream> {
                         Self::try_size_hint(depth).unwrap_or_default()
                     }
                     #[inline]
-                    fn try_size_hint(depth: usize) -> Result<(usize, ::core::option::Option<usize>), arbitrary::MaxRecursionReached> {
+                    fn try_size_hint(depth: usize) -> ::core::result::Result<(usize, ::core::option::Option<usize>), arbitrary::MaxRecursionReached> {
                         Ok(arbitrary::size_hint::and(
                             <u32 as arbitrary::Arbitrary>::try_size_hint(depth)?,
                             arbitrary::size_hint::try_recursion_guard(depth, |depth| {

